nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -275,6 +257,53 @@ in {
     (mkRenamedOptionModule
       [ "services" "nextcloud" "config" "trustedProxies" ] [ "services" "nextcloud" "settings" "trusted_proxies" ])
     (mkRenamedOptionModule ["services" "nextcloud" "extraOptions" ] [ "services" "nextcloud" "settings" ])
+
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "bucket" ] [ "services" "nextcloud" "settings" "objectstore" "arguments" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "autocreate" ] [ "services" "nextcloud" "settings" "objectstore" "autocreate" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "key" ] [ "services" "nextcloud" "settings" "objectstore" "key" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "hostname" ] [ "services" "nextcloud" "settings" "objectstore" "hostname" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "port" ] [ "services" "nextcloud" "settings" "objectstore" "port" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "useSsl" ] [ "services" "nextcloud" "settings" "objectstore" "use_ssl" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "region" ] [ "services" "nextcloud" "settings" "objectstore" "region" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" "usePathStyle" ] [ "services" "nextcloud" "settings" "objectstore" "use_path_style" ])
+    (mkRenamedOptionModule ["services" "nextcloud" "config" "objectstore" ] [ "services" "nextcloud" "settings" "objectstore" ])
+    (mkRemovedOptionModule [ "services" "nextcloud" "config" "objectstore" "enable" ] ''
+      Please move all configuration options from config.objectstore into settings.objectstore.arguments.
+      Note that two option names should be renamed: useSsl -> use_ssl and usePathStyle -> use_path_style.
+      Further instead of using secretFile and sseCKeyFile inside the objectstore scope, put both secrets
+      into the file specified in secretFile. This JSON-file should contain both secrets in this format:
+      {
+        "objectstore": {
+          "arguments": {
+            "secret": "mysecret123",
+            "sse_c_key": "mysecret123"
+          }
+        }
+      }
+      Additionally set settings.objectstore.class = "\\OC\\Files\\ObjectStore\\S3" in S3 should be used.
+    '')
+    (mkRemovedOptionModule [ "services" "nextcloud" "config" "objectstore" "secretFile" ] ''
+      Specify this objectstore secret using the secretFile option. Inside this JSON-file the secret is set
+      in this format:
+      {
+        "objectstore": {
+          "arguments": {
+            "secret": "mysecret123"
+          }
+        }
+      }
+    '')
+    (mkRemovedOptionModule [ "services" "nextcloud" "config" "objectstore" "sseCKeyFile" ] ''
```

Comment:
This option is btw still used in some places here breaking evaluation.
That said, we'll probably want to load the secretsFile with LoadCredential now that the secret options for objectstore are being removed. cc @networkException